src/common/arrstr.cpp

```diff
@@ -721,3 +732,217 @@ wxArrayString wxSplit(const wxString& str, const wxChar sep, const wxChar escape
+        {
+            fragment.type = wxStringFragment::SpaceOrPunct;
+            fragment.text = text.Left(length);
+        }
+    }
+
+    text.erase(0, length);
+    return fragment;
+}
+
+int CompareFragmentNatural(const wxStringFragment& lhs, const wxStringFragment& rhs)
+{
+    int result = 1;
+
+    switch ( lhs.type )
+    {
+        case wxStringFragment::Empty:
+            switch ( rhs.type )
+            {
+                case wxStringFragment::Empty:
+                    result = 0;
+                    break;
+                case wxStringFragment::SpaceOrPunct:
+                case wxStringFragment::Digit:
+                case wxStringFragment::LetterOrSymbol:
+                    result = -1;
+                    break;
+            }
+            break;
+
+        case wxStringFragment::SpaceOrPunct:
+            switch ( rhs.type )
+            {
+                case wxStringFragment::Empty:
+                    result = 1;
+                    break;
+                case wxStringFragment::SpaceOrPunct:
+                    result = wxStrcoll_String(lhs.text, rhs.text);
+                    break;
+                case wxStringFragment::Digit:
+                case wxStringFragment::LetterOrSymbol:
+                    result = -1;
+                    break;
+            }
+            break;
+
+        case wxStringFragment::Digit:
+            switch ( rhs.type )
+            {
+                case wxStringFragment::Empty:
+                case wxStringFragment::SpaceOrPunct:
+                    result = 1;
+                    break;
+                case wxStringFragment::Digit:
+                    if ( lhs.value >  rhs.value )
+                        result = 1;
+                    else if ( lhs.value <  rhs.value )
+                        result = -1;
+                    else
+                        result = 0;
+                    break;
+                case wxStringFragment::LetterOrSymbol:
+                    result = -1;
+                    break;
+            }
+            break;
+
+        case wxStringFragment::LetterOrSymbol:
+            switch ( rhs.type )
+            {
+                case wxStringFragment::Empty:
+                case wxStringFragment::SpaceOrPunct:
+                case wxStringFragment::Digit:
+                    result = 1;
+                    break;
+                case wxStringFragment::LetterOrSymbol:
+                    result = wxStrcoll_String(lhs.text.Lower(), rhs.text.Lower());
+                    break;
+            }
+            break;
+    }
+
+    return result;
+}
+
+} // unnamed namespace
+
+
+// ----------------------------------------------------------------------------
+// wxCmpNaturalGeneric
+// ----------------------------------------------------------------------------
+//
+int wxCMPFUNC_CONV wxCmpNaturalGeneric(const wxString& s1, const wxString& s2)
+{
+    wxString lhs(s1);
+    wxString rhs(s2);
```

Comment:
Since these are copied immediately inside the function, then the arguments can be values themselves, can't they?

Reply:
I am not sure I do understand. Did you mean to remove the const reference from the function arguments? I would rather keep the function signature same for all `wxCmpXXX() `functions.

Reply:
Ok, I didn't know about other signatures. Consistency should come first, indeed.